tests/test_TFNetworkLayer.py

```diff
@@ -3065,6 +3065,71 @@ def mask_func(source, **kwargs):
       last_var_v = var_v
 
 
+def test_reduce_mean_in_time():
+  with make_scope() as session:
+    import numpy as np
+    n_out = 5
+    config = Config({
+      "debug_print_layer_output_template": True,
+      "extern_data": {
+        "data": {"dim": n_out},
+        "classes": {"dim": n_out},
+      }})
+    net = TFNetwork(config=config, train_flag=True)
+    net.construct_from_dict({
+      "reduce_mean": {"class": "reduce", "mode": "mean", "axis": "T", "keep_dims": True, "from": ["data"],
+                      "is_output_layer": True}
+    })
+    session.run(tf.global_variables_initializer())
+    out = net.layers["reduce_mean"].output.placeholder
+    n_batch = 3
+    max_seq_len = 10
+    feed = make_feed_dict(net.extern_data.data.values(), n_batch=n_batch, n_time=max_seq_len, same_time=True)
+    v = session.run(out, feed_dict=feed)
+    input_len = feed[list(feed.keys())[0]]
+    input_data = feed[list(feed.keys())[1]]
+
+    ref = np.zeros([n_batch, n_out])
+    for batch, len in enumerate(input_len):
```

Comment:
Do not overwrite Python builtins (`len`). (PyCharm should warn about that, or not?)